docs/dev_guide/lessons/profiling_performance.rst

```diff
@@ -0,0 +1,99 @@
+
+To make things easier, there is a script included in TripalDocker that will toggle the configuration between profile mode enabled and disabled. Since it is disabled by default, we will use this script now to turn on profiling mode.
+
+.. code-block:: console
+
+  docker exec --workdir=/var/www/drupal/web/modules/contrib/tripal $containerName xdebug_toggle.sh
+
+To turn back off profiling when you are done, simple run the above script again.
+
+We also need to ensure the output directory for XDebug exists in the docker. In more recent versions of Tripaldocker this will not be needed.
+
+.. code-block:: console
+
+  docker exec --workdir=/var/www/drupal/web/modules/contrib/tripal $containerName mkdir tripaldocker/xdebug_output
+  docker exec --workdir=/var/www/drupal/web/modules/contrib/tripal $containerName chmod a+rw tripaldocker/xdebug_output
+
+Triggering Profiling of a specific page load
+----------------------------------------------
+
+There are usually browser extensions available that can trigger Xdebug. That said, it can also be triggered using a get variable and that is the approach we will use in this tutorial.
+
+For example, if you wanted to profile loading of the frontpage of your Tripal site, you would go to the following URL: http://localhost/?XDEBUG_TRIGGER.
+
+While the page is loading, XDebug is profiling the actions taken and saving the information it gleans to the /var/www/drupal/web/modules/contrib/tripal/tripaldocker/xdebug_output directory inside the docker. You can see these by looking inside that directory in the docker:
+
+.. code-block:: console
+
+  docker exec $containerName ls /var/www/drupal/web/modules/contrib/tripal/tripaldocker/xdebug_output
+
+Visualizing the output
+-----------------------
+
+Currently the results of your profiling are inside the docker container. However, we would like to bring them into your local directory to visualize them. The following commands will setup a directory to contain all the output files and then copy them from inside the container into your local directory. **Make sure you are in a directory that you want the files saved to. For example, your downloads directory.**
+
+.. code-block:: console
+
+  docker cp $containerName:'/var/www/drupal/web/modules/contrib/tripal/tripaldocker/xdebug_output' ./
+  ls xdebug_output
+
+For visualizing these files, we will use Webgrind. Since we already have docker installed, we are going to use another docker container to run webgrind. This way you don't need to install anything!
+
+.. code-block:: console
+
+  docker run --rm -v xdebug_output:/tmp -p 8081:80 jokkedk/webgrind:latest
+
+This will start the webgrind application and make it visible in your browser under port 8081. **Do not quit the open stream after running this command until you want to shut down webgrind!**
```

Comment:
```suggestion
  docker run --rm -v <xdebug_output>:/tmp -p 8081:80 jokkedk/webgrind:latest

Where `<xdebug_output>` is the name of the directory on your host system where you copied the profiling files.
This will start the webgrind application and make it visible in your browser under port 8081. **Do not quit the open stream after running this command until you want to shut down webgrind!**
```